src/lib/containers/widgets/facet-nav/FacetNav.tsx

```diff
@@ -0,0 +1,259 @@
+import * as React from 'react'
+import FacetNavPanel from './FacetNavPanel'
+import { applyChangesToValuesColumn } from '../query-filter/QueryFilter'
+import { QueryWrapperChildProps } from 'lib/containers/QueryWrapper'
+import {
+  FacetColumnResultValues,
+  FacetColumnResultValueCount,
+  FacetColumnRequest,
+  FacetColumnResult,
+} from 'lib/utils/synapseTypes'
+import { useState, useEffect } from 'react'
+import * as _ from 'lodash'
+
+export type FacetNavOwnProps = {
+  applyChanges: Function
+  loadingScreen?: React.FunctionComponent | JSX.Element
+  facetsToPlot?: string[] | undefined
+}
+
+type UiFacetState = {
+  name: string
+  isHidden: boolean
+  isExpanded: boolean
+  index?: number
+}
+
+type ExpandedFacet = {
+  facet: FacetColumnResult
+  index: number
+}
+
+type FacetNavProps = FacetNavOwnProps & QueryWrapperChildProps
+
+type ShowMoreState = 'MORE' | 'LESS' | 'NONE'
+
+const FacetNav: React.FunctionComponent<FacetNavProps> = ({
+  data,
+  getLastQueryRequest,
+  isLoadingNewData,
+  loadingScreen,
+  executeQueryRequest,
+  asyncJobStatus,
+  facetsToPlot,
+}: FacetNavProps): JSX.Element => {
+  const [facets, setFacets] = useState<FacetColumnResult[]>([])
+  const [facetUiStateArray, setFacetUiStateArray] = useState<UiFacetState[]>([])
+  const [expandedFacets, setExpandedFacets] = useState<ExpandedFacet[]>([])
+  const [isFirstTime, setIsFirstTime] = useState(true)
+
+  const request = getLastQueryRequest!()
+
+  useEffect(() => {
+    let result = data?.facets?.filter(
+      item =>
+        item.facetType === 'enumeration' &&
+        (!facetsToPlot?.length || facetsToPlot.indexOf(item.columnName) > -1),
+    )
+    if (!result) {
+      return
+    }
+    if (isFirstTime) {
+      setFacetUiStateArray(
+        result.map((item, index) => ({
+          name: item.columnName,
+          isHidden: index < 4 ? false : true,
+          isExpanded: false,
+        })),
+      )
+      setIsFirstTime(false)
+    }
+    setFacets(result)
```

Comment:
You are correct. I will put a TODO here. The reason why I'd rather not refactor this right now is that this area is likely to change depending on how we are going to handle show/hide. For example if we'd put the hidden facets in the back of the queue  rather than just hiding then in their current position we would need to maintain a queue of facets and in that case it would make sense to keep them in the state. I will add a note to make sure it does not fall through the cracks